staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -129,6 +116,19 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		audit.LogRequestObject(ae, obj, scope.Resource, scope.Subresource, scope.Serializer)
 
 		userInfo, _ := request.UserFrom(ctx)
+
+		_, name, err := scope.Namer.Name(req)
+		if err != nil {
+			if err.Error() != errEmptyName.Error() {
```

Comment:
`if err != errEmptyName`? it's odd to compare an interface's return value to a private error instance. other implementations of `Name()` wouldn't have a way to exercise this path